### PR TITLE
Tabling Rework

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request

Table slamming someone knocks down for 3 seconds instead of stunning, and does 10 brute and 40 stamina damage. Table slams while you have a martial art will destroy the table as well. Also a new sound effect for table slams.
Neck grabs followed by clicking on a table will perform a head slam, which does 40 brute damage to the head and 60 stamina distributed. This damages the table so you can only do it twice before the table breaks. Also has a chunky sound effect.

tabling OR head slamming with sleeping carp or cqc immediately destroys the table.

## Why It's Good For The Game

Infinite chain tabling is lame, as well as juggling someone between two tables while stripping every single item on them.

The intent of this is turning tabling into a powerful tool during a fight but not straight up a fight ender. Hopefully this will lead to more drawn out fights that don't just end as soon as one person is on the table.
The head slam is actually a suggestion to make the tabling slightly weaker but have a really strong alternative if you go for a full neck grab, I don't remember who suggested it but it was a good idea

## Changelog
:cl:
soundadd: New sound effect for table slams and head slams
balance: table slams knock down for 3 seconds instead of stunning for 4
balance: table slams deal 10 brute and 40 stamina damage.
add: Head slamming: neck grab someone then click a table to smash their head against it, dealing 40 brute to the head and 60 generalized stamina, as well as knocking down for 3 seconds. (stamina doesn't stack with normal damage)
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
C'mon, make it 40 brute and 40 stamina damage at the very least.

Also, chain-tabling is kind of an iconic SS13 thing, y'know? I kind of feel like we should leave it in; if someone's chain-tabling someone, they're not beating them into crit (as quickly).

**Comment by actioninja** [comment]
> iconic SS13 thing

lol

> they're not beating them into crit (as quickly).

You can beat people into crit while chain tabling them with the person having no chance of fighting back.

**Comment by py01** [comment]
One thing to fix- the PR desc and changelog desc say different amounts of brute/stamina damage.

Reasonable change, considering the table stun is long enough to cuff if you're quick.

**Comment by Dawson1917** [comment]
_ahem_

shove
floorshove
table for 30 damage
floorshove while they're knocked down from tabling
table for 30 damage
floorshove while they're knocked down from tabling
table for 30 damage
floorshove while they're knocked down from tabling
table for 30 damage

congrats, spaceman is now dead

**Comment by actioninja** [comment]
Harder to do than infinitely chaining while beating them to death and is escapable.

**Comment by Dawson1917** [comment]
> 
> 
> Harder to do than infinitely chaining while beating them to death and is escapable.

delete floorshoves

**Comment by actioninja** [comment]
I agree.

**Comment by SkeletalElite** [comment]
I think it'd be better if it did just stamina damage and not brute. I think it's better off the act of tabling itself stay non lethal.

**Comment by necromanceranne** [comment]
This is going to get really silly with sleeping carp and leather golems.

**Comment by Twaticus** [comment]
tablestunning now is bullshit and leaves the person being tabled basically powerless so i think this is a good change, minus the brute damage. 

giving damage kind of ruins the fun of tabling someone since most of the time youre not trying to actually hurt them. 
if you want to use a table to take someone out, you should have to also harm them instead of just relying on constant tabling

**Comment by Akrilla** [comment]
add more cool unarmed fighting stuff

**Comment by necromanceranne** [comment]
At 30 brute and 30 stamina it's really, really fucking powerful against someone who is already on the floor given kicking potency. The guy who gets tabled is going to be in a bad way and it's a shame it has to be a lethal killing tool now.

I do have concerns about the possibility of still getting someone into a death spiral from being shoved into a table, stunned, tabled, shoved down again and then repeat for even more damage to the point where getting kicked will likely finish you off. Getting grabbed while on the floor is much more reliable with a ton of damage than if you were at full health, so I don't know if I'm okay with 30 brute over just making it entirely stamina damage.

Also again, Sleeping Carp boys will literally smash you over and over into a table at esword damage.

**Comment by Dawson1917** [comment]
You could make it break non-reinforced (or reinforced too to prevent RnD desk / genetics desk chainstunning) tables so you can't chaintable someone outside of places like the bar, and then the 30 brute & stam damage doesn't seem so bad.

But being able to re-tablestun someone who hasn't even recovered from the stun has always been very silly. Especially on the same table.

File: code/game/objects/structures/tables_racks.dm [hunk 1]
```diff
@@ -126,7 +126,10 @@
 		pushed_mob.pass_flags &= ~PASSTABLE
 	if(pushed_mob.loc != loc) //Something prevented the tabling
 		return
-	pushed_mob.Paralyze(40)
+	pushed_mob.Knockdown(20)
+	pushed_mob.adjustBruteLoss(30)
```
Comment (nemvar) [inline code/game/objects/structures/tables_racks.dm]: It would be cool if this respected armor.

**Comment by GuyonBroadway** [comment]
I still think if you neckgrab someone and do *flip you should suplex them into and through an adjacent table or down a disposals chute but thats just me.

**Comment by Shapsy** [comment]
Wait what if I just want to playfully throw someone onto a table because they insulted me in the bar? Doing 30 brute would make it look like I want to outright murder them, I don't see why shoving someone onto a table should do more harm to them than a fire-axe to the head

**Comment by carlarctg** [comment]
tabling someone on harm intent should deal full damage, on disarm it should deal stamina damage, and on help intent no damage

**Comment by Ratvar** [comment]
"Hopefully this will lead to more drawn out fights"
adds 30 brute damage on top of an already loosing side
why stamina isn't enough?

**Comment by Shadowflame909** [comment]
this is a full ass e-sword

replace tables with energy tables and then we'll talk

**Comment by optimumtact** [comment]
make the headslam damage the table so you can only do it twice or something before you need more table

**Comment by carlarctg** [comment]
people here you do realize stamina and normal.damage dont stack anymore right

**Comment by actioninja** [comment]
@carlarctg honestly I forgot that while making this

File: code/game/objects/structures/tables_racks.dm [hunk 2]
```diff
@@ -126,15 +129,36 @@
...
+	pushed_mob.apply_damage(40, BRUTE, BODY_ZONE_HEAD)
+	pushed_mob.apply_damage(60, STAMINA)
+	take_damage(50)
+	if(user.mind?.martial_art)
+		deconstruct(FALSE)
+	playsound(pushed_mob, "sound/effects/tableheadsmash.ogg", 90, TRUE)
+	pushed_mob.visible_message("<span class='danger'>[user] smashes [pushed_mob]'s head against \the [src]!</span>",
+								"<span class='userdanger'>[user] smashes your head against \the [src]</span>")
+	log_combat(user, pushed_mob, "head slammed", null, "against [src]")
+	if(!ishuman(pushed_mob))
```
Comment (nemvar) [inline code/game/objects/structures/tables_racks.dm]: Why the ishuman check before the mood event signal? This looks like legacy mood code.

File: code/game/objects/structures/tables_racks.dm [hunk 3]
```diff
@@ -126,15 +129,36 @@
...
 		return
 	var/mob/living/carbon/human/H = pushed_mob
 	SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "table", /datum/mood_event/table)
 
+/obj/structure/table/proc/tableheadsmash(mob/living/user, mob/living/pushed_mob)
+	pushed_mob.Knockdown(30)
+	pushed_mob.apply_damage(40, BRUTE, BODY_ZONE_HEAD)
+	pushed_mob.apply_damage(60, STAMINA)
+	take_damage(50)
+	if(user.mind?.martial_art)
```
Comment (nemvar) [inline code/game/objects/structures/tables_racks.dm]: Every mind has a default martial arts datum. Tables will ALWAYS get destroyed

```
/datum/mind/New(var/key)
	src.key = key
	soulOwner = src
	martial_art = default_martial_art
```

**Comment by mrhugo13** [comment]
19 Thumbs down and 13 Thumbs up?

Now i *know* that this is a good change!

**Comment by Ty-the-Smonk** [comment]
Carp should have a interaction if you Insta throw someone onto a table

**Comment by Difrit** [comment]
This will create further problems with escalation, as tabling someone to try and get away from them while they're bothering you will now give them full right to mess your day up.

Also this is literally better than any other weapon available to the entire crew and most antags, was any thought put into those damage values?

**Comment by Fhaxaris** [comment]
Right now the tabling system is utterly broken. You can literally go from table to strangle.

1. Table them and open their inventory
2. While they are tabled neck grab and table again.
3. Grab up to an aggressive grab before the progress bar completes.
4. When they are neck grabbed, grab again.
5. Beat them into crit/full-strip while they only had a 5% to 15% chance to escape.

I don't think very many people know this. They only have a 5% chance of breaking out of it during the neck grab phase, and It lets you beat someone to death while they can't even fight back or call for help.

**Comment by carlarctg** [comment]
you don't need to neck grab? shove into a wall and permastun is real though

**Comment by Ty-the-Smonk** [comment]
> This will create further problems with escalation, as tabling someone to try and get away from them while they're bothering you will now give them full right to mess your day up.
> 
> Also this is literally better than any other weapon available to the entire crew and most antags, the normal tabling causes more stam damage than most stamina-based weapons and the necktable does the same as batons afaik, was any thought put into those damage values?

That’s kinda the point? If you are tabling someone to steal their shit you should be valid salad right there and then, this is just a nerf to tiders

**Comment by ExcessiveUseOfCobblestone** [comment]
> Right now the tabling system is utterly broken. You can literally go from table to strangle.
> 
> 1. Table them and open their inventory
> 2. While they are tabled neck grab and table again.
> 3. Grab up to an aggressive grab before the progress bar completes.
> 4. When they are neck grabbed, grab again.
> 5. Beat them into crit/full-strip while they only had a 5% to 15% chance to escape.
> 
> I don't think very many people know this. They only have a 5% chance of breaking out of it during the neck grab phase, and It lets you beat someone to death while they can't even fight back or call for help.

are you talking about post this PR or pre this PR? You could easily stunlock someone w tabling prePR.

**Comment by Fhaxaris** [comment]
> this PR or pre this PR?

It is doable pre the PR and post the PR, the method of doing it is slightly different. The pre PR issue was that this let you neck grab to full strangle while they were stunned from the tabling while they only had a max of 3 chances to resist. The post PR issue is very similar, it allows you to neck grab while you are tabling them which means you can neck slam right after tabling.

They both stem from the same issue, that the neck grab animation/bar doesn't go away after tabling, so you can grab up to an aggressive grab and the neck grab will complete.

**Comment by PKPenguin321** [comment]
It will actually be even stronger post-PR due to how grabs interact with stamina

**Comment by Takov-Mala** [comment]
Doesn't matter if it's 30 or 40, both are **insane** and bizarre amounts of damage for this to do. This "nerf" does the exact opposite of it's intention. Very few people are okay with it doing any damage at all, chainstunning is dumb yes but this is infinitely worse. This doesn't suit the gameplay of TGstation in the slightest. 

Please, stop. This is the kind of stuff that makes playing the game worse for everyone.

**Comment by ExcessiveUseOfCobblestone** [comment]
> This doesn't suit the gameplay of TGstation in the slightest.

what does this mean?

**Comment by carlarctg** [comment]
> > This doesn't suit the gameplay of TGstation in the slightest.
> 
> what does this mean?

it means this ruins the culture of tg

i'm not a fan of tabling doing damage ALWAYS, maybe only on harm intent?
it doing 30 damage is totally okay because 30 damage is 30 STAMINA damage which does literally a whole load of nothing by itself

**Comment by ExcessiveUseOfCobblestone** [comment]
https://www.oranges.net.nz/tglist.html

**Comment by optimumtact** [comment]
I don't even understand why tabling has to do any damage other than stamina damage at all

**Comment by Takov-Mala** [comment]
> 
> 
> > This doesn't suit the gameplay of TGstation in the slightest.
> 
> what does this mean?

Different servers often have different gameplay, this is basic knowledge.
Pretty sure I had a longer but still calm response here previously, but I guess it was deleted?  Why even ask if you're just going to silence the explanation? Christ dude.



> 
> 
> I don't even understand why tabling has to do any damage other than stamina damage at all

Not to mention just stunning and agg tabling someone is now more effective than most traitor gear.

**Comment by ExcessiveUseOfCobblestone** [comment]
there's no stealth delete option (for me), either you deleted it yourself or it didn't go through.

> Different servers often have different gameplay, this is basic knowledge.

And that gameplay is malleable via code.

**Pull request merged**